### PR TITLE
Chapter "Online-Quotes" created.

### PR DESCRIPTION
**Pull request by CWehli**
* Description of the installation and configuration of F::Q (moved from
Help_ch_Account-Actions-xml).
* Some passages taken from guide/C/ch_invest.xml in PR #192, commit 7e4e560 (Author
usamimikamen-sk)
* adjust some internal references

### CODE REVIEW

**Comment by fellen** [comment]
I wonder if we should move to here also the preparation to use the SQL backend from guide and wiki.

**Comment by jralls** [comment]
> I wonder if we should move to here also the preparation to use the SQL backend from guide and wiki.

Most of guide sections 2.3 through 2.7 belong in help instead, but not in a chapter called "Suggested Applications".

Also "Suggested Applications" is not a good name for a chapter explaining how to install AQB and F::Q because those aren't applications, they're libraries.

The only info in the wiki at present about the SQL backends details the database schema. That doesn't really belong in either the guide or help any more than the XML schema does.

**Comment by fellen** [comment]
> Also "Suggested Applications" is not a good name for a chapter explaining how to install AQB and F::Q because those aren't applications, they're libraries.

Install and configure optional dependencies?

**Comment by fellen** [comment]
> The only info in the wiki at present about the SQL backends details the database schema. That doesn't really belong in either the guide or help any more than the XML schema does.

I started mentioning the wiki after I saw several outdated sections in the docs.

**Comment by jralls** [comment]
> Install and configure optional dependencies?

Maybe. That works for F::Q, but either GnuCash is built with AQB or it's not. If not the only way to change it is to install AQB and rebuild. I don't think that's the level of user we're addressing in the docs.
Besides, I don't think anyone distributes GnuCash without AQB.

**Comment by fellen** [comment]
Setup of helper tools?

**Comment by jralls** [comment]
Optional Dependencies seems better than Helper Tools, but since there's only one how about Setup Finance Quote?

**Comment by fellen** [comment]
> Most of guide sections 2.3 through 2.7 belong in help instead, but not in a chapter called "Suggested Applications".

We could also reorganize Customizing GnuCash as:
Internal: (current content)
External: Aqb, dbi, F::Q…

**Comment by jralls** [comment]
Guide sections 2.3 through 2.7 would I think be spread out through help chapters 3 - 7.
AQB and F::Q aren't really customizations, they each deserve their own chapters. 

DBI is an implementation detail. The only thing a user needs to know is that they need it if they want to use SQL and sometimes it has to be installed separately in the package manager. Most users are probably better off not even knowing about SQL or that there's a choice of ways to store their book.

**Comment by CWehli** [comment]
> Guide sections 2.3 through 2.7 would I think be spread out through help chapters 3 - 7.
> AQB and F::Q aren't really customizations, they each deserve their own chapters.

In fact, we have not yet described the topics from 2.4 to 2.6 in the help. 
I think the new thematic chapter would fit behind chapter 4 (GUI) and I imagine "File Operation" as the title.

The contents of 2.3 and 2.7 are already in the help and could also be removed from the guide.

If, in your opinion, there is no need for further explanations for AqB and DBI, then I will better rename the title I have chosen to "Setup Finance Quote".

**Comment by fellen** [comment]
> If, in your opinion, there is no need for further explanations for AqB and DBI, then I will better rename the title I have chosen to "Setup Finance Quote".

About AqB we have currently a stub in https://code.gnucash.org/docs/C/gnucash-help/ch_AccRegTools.html several wiki pages. FinTS and Paypal in de, directOFX in en.
IMHO Tools should primary describe Action->OnlineActions and the new chapter Tools->Online Banking Setup.

**Comment by jralls** [comment]
> If, in your opinion, there is no need for further explanations for AqB and DBI, then I will better rename the title I have chosen to "Setup Finance Quote".

Just **Finance Quote**. It should cover setup, of course, but also how to configure commodities of various types for retrieval, how to retrieve prices, what the various errors mean and what to do about them, and so on.

> About AqB we have currently a stub in https://code.gnucash.org/docs/C/gnucash-help/ch_AccRegTools.html several wiki pages. FinTS and Paypal in de, directOFX in en.
IMHO Tools should primary describe Action->OnlineActions and the new chapter Tools->Online Banking Setup.
[Edit:] As there is no need to have OFX in german or FinTS in english, I would prefer to use ITS for that.

I don't think that the menus make a particularly useful way to organize the manual. There can be a chapter about the menu structure that has cross-references to chapters, but chapters should be organized thematically. For example we could have an Accounts page chapter that explains the features of the accounts page like adding and removing columns, editing individual items directly, the Account Edit dialog, and other tasks that one might do on that page. We could have chapters for Register and Report pages, and then a chapter describing each of the built-in reports and a separate chapter on searching the database that's linked from the Accounts and Register chapters.

File: manual/de/ch_Online-Quotes.xml [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+      <step>
+        <simpara>Führen Sie den Befehl <command>which gnc-fq-update</command> aus, um zu überprüfen, ob das
+          Programm <filename>gnc-fq-update</filename> in der PATH Umgebungsvariablen eingetragen
+          ist.
+          <footnote>
+            <simpara>Wenn Sie die von Ihrer Distribution bereitgestellt &app; Pakete installiert haben, sollte
+              <filename>gnc-fq-update</filename> bereits in Ihrem PATH sein. Die Aktualität ihrer
+              Distribution können Sie unter
+              <ulink url="https://repology.org/project/perl:finance-quote/versions"><citetitle>&app-fq;-
```
Comment (fellen) [inline manual/de/ch_Online-Quotes.xml:58]: Sollten wir für `https://repology.org/` nicht eine Entität erstellen?

**Review by fellen — COMMENTED** [review]
So, heute habe ich hoffentlich endlich kapiert, was Cmake mit ninja wohin baut oder installiert.
Guck mal, ob in meinen Kommentaren noch was brauchbares ist.

File: manual/C/ch_Online-Quotes.xml [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    constantly. &app; has the ability to automatically download the most recent price for your
+    commodities using the Internet. This is accomplished through the &app-perl; module &app-fq;,
+    which must be installed in order to activate this feature.
+  </para>
+
+  <sect1 id="online-quotes-install">
+    <title>Installing &app-fq;</title>
+
+    <para>To determine if the &app-perl; module &app-fq; is already installed on your system, type
+      <command>perldoc Finance::Quote</command> in a terminal window and check to see if there is
```
[the remark this answers is not in the document]
Comment (CWehli) [reply manual/C/ch_Online-Quotes.xml:29]: You are right, this description is not completely valid for all OS. The requirements for a successful installation are also not yet listed.

[the remark this answers is not in the document]
Comment (fellen) [reply manual/C/ch_Online-Quotes.xml:29]: IMHO we should in the text only have Finance-Quote and Perl
AFAIK cpan is part of Perl and the rest are dependencies of F::Q and should be declared there.
Any hotfixes like missing Mozilla's certificate authorities should be at best a footnote.
Perhaps such floating targets are better kept in the wiki.

[the remark this answers is not in the document]
Comment (CWehli) [reply manual/C/ch_Online-Quotes.xml:29]: > IMHO we should in the text only have installing Finance-Quote and Perl

I think we don't need to describe the installation of Perl, but we need to mention it as a prerequisite.

[the remark this answers is not in the document]
Comment (fellen) [reply manual/C/ch_Online-Quotes.xml:29]: > I think we don't need to describe the installation of Perl, but we need to mention it as a prerequisite.

How about a stream lined version of https://wiki.gnucash.org/wiki/Online_Quotes#Perl
Windows (=GnuCash 2.5 and more recent; footnote=GnuCash 2.4 and before)

**Review by fellen — APPROVED** [review]
As it is a big improvement, I will commit it now—as christmas gift.
@
I have attached a few suggestions to thevGeran version for later improvement.

File: manual/C/ch_Finance-Quote.xml [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        <simpara>Run <command>crontab -e</command>.
+        </simpara>
+      </step>
+
+      <step>
+        <para>Add the following line to your crontab:<informalexample>
+<screen>0 16 * * 5 gnucash-cli --quotes get <replaceable>${HOME}/gnucash-filename</replaceable> &gt; /dev/null 2&gt;&amp;1</screen>
+          </informalexample><warning>
+            <para>If there is no X session running on your computer at the time of the quote request, you must do the
+              call as follows instead:<screen>0 16 * * 5 env `dbus-launch` sh -c 'trap "kill $DBUS_SESSION_BUS_PID" EXIT; gnucash-cli --quotes get <replaceable>${HOME}/gnucash-filename</replaceable> &gt; /dev/null 2&gt;&amp;1</screen>
```
Comment (fellen) [inline manual/C/ch_Finance-Quote.xml:228]: entry [in crontab], not call

**Pull request merged**